lib/media/streaming_engine.js

```diff
@@ -1168,6 +1194,15 @@ shaka.media.StreamingEngine = class {
           mediaState.clearBufferSafeMargin);
       return;
     }
+    // Handle deferred closeSegmentIndex() operations.
```

Comment:
Please move this to a private method that can also be called when tearing down StreamingEngine (when there would be no more onUpdate_ calls).

Reply:
Done.